build/listing-loader.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const { Compilation } = require('webpack');
+
+require('../src/common/tools/setup-ts-in-node');
+
+const { crawl } = require('../src/common/tools/crawl.ts');
+const { assert } = require('../src/common/framework/util/util.ts');
+
+// See https://webpack.js.org/api/loaders/.
+// Module loader that replaces an import with a module that contains the generated
+// listing for a test suite. Assumes the listing is at /path/to/{suite}/listing.ts.
```

Comment:
I was thinking for the moment we might still require a build step when new files are added. OTOH, if we have code to generate listing.ts running server-side (which I assume this is?) then should we just do all the TS compilation server-side...?

Reply:
I didn't actually verify this yet, but Webpack's watcher should regenerate the listing on changes. If it doesn't we should be able to set it up to do that.

But yea you're right we could probably make all the compilation happen server side without the need for service worker. It'll probably be faster that way too. 

It might be cleaner to just write our own http server for this.

Reply:
I'm going to try our own http server. It avoids bundling shenanigans and is less memory and is much simpler, especially if we intend for this to be dev only.

Reply:
Sounds good. I really would like to go the service worker route for WPT someday, but given we have no way to do this now with a reasonable file size, this does seem more straightforward.

Reply:
I was pretty excited about this, but now seeing that the compiles take tens and sometimes 100-something milliseconds (maybe cause the browser is busy doing something else like fetching, etc.) it seems like it potentially could be rather detrimental for test runner performance and increase run times on CI a lot. Maybe the code structure could be better, because it looks like with the service worker, fetches and compiles aren't happening in parallel. If we did Promise.all of all the dynamic `import`s instead of `await`ing each, maybe this would be less of a problem.

How do you imagine we would go the service worker route for WPT while maintaining a reasonable file size?
 - WPT gains a fetch deps + build step? (might as well just compile the TS here)
 - We use a tiny TS compiler?

If we want WPT service worker to happen in the nearish-term, maybe we should stick with service worker on standalone so most of the code is there.

Reply:
another important bit here is that service worker won't work if you serve the CTS from a remote server. It's explicitly allowed for `localhost`, but if you try to get at it from some random IP address, service worker isn't available - unless you run with --disable-web-security

Reply:
We can definitely get the fetches to happen in parallel. Though I assume by default the service worker won't actually parallelize the synchronous TS compilation?

My only idea for WPT is to use a tiny TS compiler (I haven't been able to find any such thing, even though it's theoretically possible), or get approval to drop a big dep into WPT (no idea how likely).

Good to know about the `localhost` thing, is that it would work on localhost or https but not http?

Reply:
Yup, exactly

Reply:
Server version is in #386